### PR TITLE
Migrate to PySPNEGO

### PR DESCRIPTION
**Pull request by clubby789**
The ntlm_auth library is deprecated in favour of pyspnego. This commit
migrates to this library, patching some methods to ensure dependent code
will continue to work.

### CODE REVIEW

**Comment by fidelitas** [comment]
Hi, thank you for your work here!
May I ask what's the status of your PR?

thank you and best regards

**Comment by clubby789** [comment]
The status of my PR? As far as I'm aware it's fully working, and I'm waiting on a maintainer to review it

**Comment by srmo** [comment]
Ahoi! Can we get an update on this? Otherwise I'd dive into pyspnego now.

**Review by nitzmahone — APPROVED** [review]

**Comment by jborean93** [comment]
Thanks for the PR @clubby789, I've slightly modified it to support channel binding tokens and not use the internals for the wrapping/unwrapping code.

**Comment by herrbenesch** [comment]
🎉  thanks @clubby789 and @jborean93

**Comment by matthias-reiff** [comment]
Thanks a lot! 
Works like a charm and I don't have to tweak my openssl config unnecessarily!

**Pull request merged**
